core/src/main/scala/sbt/librarymanagement/VersionNumber.scala

```diff
@@ -27,6 +27,10 @@ final class VersionNumber private[sbt] (
     case _                => false
   }
 
+  def satisfies(selector: String): Boolean = {
```

Comment:
"satisfies" might be too strong since Ivy/Coursier/Maven have their own implementations, and also many of Scala or Java versions are not SemVer. A better name might be `matchesSemVer`, and let it accept `selector: SemanticSelector`?